From 518a8593d42053bfe77fc0827cde99bbcc89579d Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 27 Apr 2016 10:41:51 +0300
Subject: [PATCH 0374/2241] sdio: dts: ap806: add regulator for voltage change
 for SDIO cards

Currently we have an issue with SD/MMC when change the voltage,
this patch add a regulator node as place holder

Change-Id: Iccf591b5a121e6097733434a320b868055ecc3da
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29397
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
(cherry picked from commit 5fe29a960ee74633ea41baa2e36e0e040daea94c)
Reviewed-on: http://vgitil04.il.marvell.com:8080/29566
---
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi | 8 ++++++++
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi   | 8 ++++++++
 2 files changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 1f6a06d..0ae6a8e 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -58,6 +58,14 @@
 
 	ap806 {
 		config-space {
+			sd_mmc_reg0: regulator@0 {
+				gpios = <&gpio0 19 GPIO_ACTIVE_HIGH>;
+				gpios-states = <0>;
+				states = <1800000 0x1
+					3300000 0x0>;
+				enable-active-high;
+				status = "disabled";
+			};
 			sdhci@6e0000 {
 				bus-width = <4>;
 				non-removable; /* For now, need to configure the GPIO for card detect */
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 1d478cb..ab08af5 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -242,6 +242,14 @@
 					<GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
 			};
 
+			sd_mmc_reg0: regulator@0 {
+				compatible = "regulator-gpio";
+				regulator-name = "sd-mmc-gpio-supply";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+			};
+
 			serial@512000 {
 				compatible = "snps,dw-apb-uart";
 				reg = <0x512000 0x100>;
-- 
2.7.4

